geode-redis/src/main/java/org/apache/geode/redis/internal/executor/hash/GeodeRedisHashSynchronized.java

```diff
@@ -89,6 +89,11 @@ public int hdel(List<ByteArrayWrapper> subList) {
       numDeleted.set(oldHash.size() - newHash.size());
       return newHash;
     });
+
+    if (hgetall().size() == 0) {
+      RedisDataType type = context.getKeyRegistrar().getType(key);
```

Comment:
It seems like this code should only unregister if the current type is REDIS_HASH. I thought you were unregistering here but does getRegionProvider().removeKey change the keyRegistar? I will talk with Sarah about this. I might just not understand this code

Reply:
Yes, the removeKey method unregisters the key from the keyRegistrar in addition to deleting it.